2048/2048/src/main/java/com/appcoins/eskills2048/activity/FinishGameActivity.java

```diff
@@ -27,23 +38,28 @@
   public static final String SESSION = "SESSION";
   public static final String WALLET_ADDRESS = "WALLET_ADDRESS";
   public static final String USER_SCORE = "USER_SCORE";
+  private static final Long GET_ROOM_PERIOD_SECONDS = 3L;
 
   private ActivityFinishGameBinding binding;
   private FinishGameActivityViewModel viewModel;
   private final static int PARTY_POPPER_EMOJI_UNICODE = 0x1F389;
   private final static int PENSIVE_FACE_EMOJI_UNICODE = 0x1F614;
   private CompositeDisposable disposables;
+  private RecyclerView recyclerView;
+  private PlayerRankingAdapter adapter;
+  private Boolean roomFinished = false;
```

Comment:
In a highly concurrent environment, we should avoid side effects as much as possible. There is no need for a boolean variable to check if in the chain you have the room